Helper/Data.php

```diff
@@ -431,25 +389,131 @@ public function cronAbandonedcart($orders)
         }
     }
 
-    private function alertCustomer($row, $fields)
+    /**
+     * Formats cart fields to standard abandoned cart template fields used in Smaily.
+     *
+     * @param array $row            Cart information.
+     * @param array $selectedFields Fields selected for template.
+     * @return array                Array of 'customer_data' and 'products_data' sections.
+     */
+    private function prepareCartData($row, $selectedFields)
     {
-        $responderProduct = [];
+        // Populate customer data section.
+        $customerData = [
+            'email' => $row['customer_email']
+        ];
 
+        if (in_array('first_name', $selectedFields, true)) {
+            $customerData['first_name'] = $row['customer_firstname'];
+        }
+
+        if (in_array('last_name', $selectedFields, true)) {
+            $customerData['last_name'] = $row['customer_lastname'];
+        }
+        // Populate product data section.
+        $productsData = [];
+        $objectManager = \Magento\Framework\App\ObjectManager::getInstance();
+        $fieldsAvailable = $this->getAllAbandonedCartFields();
         foreach ($row['products'] as $product) {
             $_product = [];
-            foreach ($product as $field => $val) {
-                if ($field === 'name' || in_array($field, $fields, true)) {
-                    $_product['product_' . $field] = $val;
+            foreach ($fieldsAvailable as $field) {
+                if (in_array($field, $selectedFields)) {
+                    switch ($field) {
+                        case 'first_name':
+                        case 'last_name':
+                            // Skip customer fields.
+                            break;
+                        case 'qty':
+                            // Transform qty to quantity.
+                            $_product['product_quantity'] = $product[$field];
+                            break;
+                        case 'description':
+                            $productObject = $objectManager->
+                                create('Magento\Catalog\Model\Product')->
+                                load($product['product_id']);
+                            $description = $productObject->getDescription();
+                            $_product['product_description'] = htmlentities($description);
```

Comment:
Should the description be converted to entities or stripped of tags or somehow else modified before we send it to our API?

A current example from the demo store. Value of the `product_description` field:
```
'<p>The Breathe Easy Tank is so soft, lightweight, and comfortable, you won\'t even know it\'s there -- until its high-tech Cocona&reg; fabric starts wicking sweat away from your body to help you stay dry and focused. Layer it over your favorite sports bra and get moving.</p>
<p>&bull; Machine wash/dry.<br />&bull; Cocona&reg; fabric.</p>',
```

Reply:
If I remember correctly we did go over this at some point with some modules/extensions/plugins. And if I am not mistaken then we decided that HTML should be left as-is and wrapped in CDATA tags:
```xml
<description><![CDATA[<p>This is my product description.</p>]]></description>
```

Reply:
I think this was regarding the RSS-feed XML-structure

Reply:
Of course... I got the two mixed up.

Come to think of it - why would the product description even be needed in abandoned cart emails? Usually it contains only the list of products added to cart (even stores own shopping cart doesn't show product description).

If product data is still needed. I would still send it as is (though WooCommerce plugin applies `htmlspecialchars` to description), because Smaily accepts the payload as is an pipes it to the template. Otherwise `<p>` would be rendered to template as `&lt;p&gt;`.

Reply:
Changed description to use `htmlspecialchars()` function instead. https://github.com/sendsmaily/smaily-magento-extension/pull/33/commits/9dbdaa5eac83a04fae5e36206445be298ffa0609